### PR TITLE
Configurable favorite sort options

### PR DESCRIPTION
**Pull request by ThoWagen**
As discussed in #3884 this PR adds the possibility to configure the favorite list's sort options.

@demiankatz concerting your comment

> I wonder if the solution should involve a str_replace in one direction or the other (e.g. normalize spaces to underscores), so legacy inconsistent values continue to work. 

I added the alias `sort_year asc = "sort_year_asc"` so outdated configurations should still work.

Also, there are a lot of automatic cs fixes by my IDE included. Those are removing spaces at the end of lines and adding empty lines at the end of files. I figured that they might actually be an improvement. But please tell me if you don't want them and I will revert those changes.

### CODE REVIEW

File: module/VuFind/tests/fixtures/configs/1.1/Summon.ini [hunk 1]
```diff
@@ -143,15 +143,15 @@ TableOfContents     = adv_search_toc
...
-; of the equal sign are text that will be run through the translation module and 
+; or the name of a Summon index to use for sorting; asc and desc modifiers may be
+; used in combination with index names, but not relevance.  Values on the right
+; of the equal sign are text that will be run through the translation module and
 ; displayed on screen.
 [Sorting]
 relevance = sort_relevance
 PublicationDate:desc = sort_year
-PublicationDate:asc = "sort_year asc"
+PublicationDate:asc = sort_year_asc
```
Comment (demiankatz) [inline module/VuFind/tests/fixtures/configs/1.1/Summon.ini]: The config files under fixtures are historical legacy configurations used for upgrade tests, and should not be changed. Do you mind reverting these few files?

Given that we no longer support upgrading from version 1 anyway, and I plan to remove the infrastructure supporting it in release 11, some of them may go away soon anyway.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @ThoWagen, I haven't tried this out yet, but it looks like it's off to a great start. One request regarding the config files, though...

File: module/VuFind/tests/fixtures/configs/1.1/Summon.ini (hunk 1 above)
Comment (ThoWagen) [reply module/VuFind/tests/fixtures/configs/1.1/Summon.ini]: Sorry, I was not aware of that. I reverted the changes there. (Only the fixture files under 'configs/1.*' are legacy right?)

Comment (demiankatz) [reply module/VuFind/tests/fixtures/configs/1.1/Summon.ini]: The worldcat fixture is also legacy (and will soon be deleted as well when the WorldCat v1 API is taken offline and all related code will need to be removed).

Comment (ThoWagen) [reply module/VuFind/tests/fixtures/configs/1.1/Summon.ini]: done

**Review by demiankatz — APPROVED** [review]
Thanks, @ThoWagen, this looks good to me!

**Pull request merged**